src/tools/tracejob.c

```diff
@@ -405,7 +405,12 @@ parse_log(FILE *fp, char *job, int ind)
 		while(buf_size == strlen(buf) + 1) {
 			buf_size *= 2;
 			tbuf = (char*)realloc(buf, (buf_size + 1) * sizeof(char));
-			if(tbuf == NULL || (buf = tbuf) && fgets(buf + strlen(buf), buf_size/2 + 1, fp) == NULL){
+			if (!tbuf) {
```

Comment:
I might be wrong, but it seems like you are changing the functionality here. Shouldn't it be:
if (!tbuf) {
---->buf = tbuf;
---->if fgets(buf + strlen(buf), buf_size/2 + 1, fp) == NULL) {
---->---->break_fl = 1;
---->---->break;
---->}
} else {
---->break_fl = 1;
---->break;
}

Reply:
The code was not very readable in the state it was in. The goal is to increase the size of buf until it is large enough to hold a long line by doubling its size as many times as required. You need to be careful with realloc because when it returns NULL the original buffer is left alone. If we lose track of it, we'll leak memory. Setting buf to tbuf when we know tbuf is NULL would do just that.

Reply:
ah, you are right, I read it incorrectly, my bad.